### ISSUE TITLE
DbUnitChecker: gets confused by the word "field" in comments of DBs

### ISSUE THREAD

**Issue by Tom-Willemsen** [ISISComputingGroup/IBEX#4173]
As a developer, I don't want the DbUnitChecker to fall over if the word "field" is present in comments in an epics DB.

Example (from IPS):

```
record(calc, "$(P)FIELD:USER") {
    field(CALC, "(C=0||C=2)?A:B")  # If heater is present and switched off, display persistent field, else display PSU field
    field(ASG, "READONLY")
}
```

Gives an error:

```
AssertionError: Multiple fields on PVs in C:\Instrument\Apps\EPICS\ioc\master\IPS\IPS-IOC-01App\Db\ips.db
   -> Multiple instances of fields ASG on $(P)FIELD:USER
```

This is because it attempts to interpret the word "field" (in the comments) as an epics declaration, and then comes accross the `ASG` declaration multiple times

Easy version of this ticket: fix the regex based parser so that it doesn't misinterpret comments, for example by pre-filtering out anthing following an unescaped `#` character until the end of the line.

Better (but also harder) version of this ticket: replace the regex-based parser with the recursive descent parser which is implemented in https://github.com/ISISComputingGroup/DbChanges , which is much more likely to be robust

**Comment by LiamPanchaud** [comment]
All good for the most part! See comments on pull request for a small change

**Comment by SophieKSTFC** [comment]
Changes needed for rework will be implemented in the unit test refactoring as it comes under testing the db parser (see ticket ISISComputingGroup/IBEX#2450 )

**Comment by DominicOram** [comment]
Could we pull this back in now that #2450 is done?

**Comment by SophieKSTFC** [comment]
The issue should have been closed when #2450  was closed as the rework was completed as part of that ticket.
